processing/src/main/java/io/druid/query/groupby/GroupByQueryEngine.java

```diff
@@ -312,12 +314,113 @@ public RowIterator(GroupByQuery query, Cursor cursor, ByteBuffer metricsBuffer,
       dimensionSpecs = query.getDimensions();
       dimensions = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
       dimNames = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
+
       for (int i = 0; i < dimensionSpecs.size(); ++i) {
         final DimensionSpec dimSpec = dimensionSpecs.get(i);
-        final DimensionSelector selector = cursor.makeDimensionSelector(dimSpec.getDimension());
-        if (selector != null) {
-          dimensions.add(selector);
+        final String dimension = dimSpec.getDimension();
+        if(dimension.equals(Column.TIME_COLUMN_NAME)) {
```

Comment:
I'm not sure which one is better frankly. The query engine has better assumptions regarding how the data is scanned, so it can potentially make better decisions on how to optimize, based on the storage adapter capabilities. The Storage adapter may be able to optimize certain things, but would have to handle the more general case, since it doesn't know what the query engine might do.

We can push down all extraction functions to the storage adapter level, if you feel this would be more consistent?